kani-compiler/src/kani_middle/attributes.rs

```diff
@@ -91,6 +101,7 @@ pub(super) fn check_attributes(tcx: TyCtxt, def_id: DefId) {
             KaniAttributeKind::Unstable => attrs.iter().for_each(|attr| {
                 let _ = UnstableAttribute::try_from(*attr).map_err(|err| err.report(tcx));
             }),
+            KaniAttributeKind::Ensures | KaniAttributeKind::Requires => (),
```

Comment:
The validation logic should go here, not inside `extract_harness_attributes`.

Reply:
I'm delaying doing this until the attributes are stable(ish) #2595